### PR TITLE
Fix brew-copy.sh and specify package versions and Homebrew commits in the release message

### PR DESCRIPTION
**Pull request by mbautin**
Example release: https://github.com/mbautin/brew-build/releases/tag/20191018T024308-linux

### CODE REVIEW

**Comment by ttyusupov** [comment]
Example release only has one package in release notes, is this expected?
```
Included package versions:
pachelf 0.10
```

File: brew-common.sh [hunk 1]
```diff
@@ -20,8 +20,33 @@ if [[ $BASH_SOURCE == $0 ]]; then
...
+
 declare -i -r ABS_PATH_LIMIT=85
 
+if [[ $OSTYPE == linux* ]]; then
+  readonly YB_WHATBREW=linuxbrew
+  readonly YB_WHATBREW_CAPITALIZED=Linuxbrew
+  readonly YB_OS_FAMILY=linux
+elif [[ $OSTYPE == darwin* ]]; then
+  readonly YB_WHATBREW=homebrew
+  readonly YB_WHATBREW_CAPITALIZED=Homebrew
```
Comment (ttyusupov) [inline brew-common.sh]: Will `YB_BREW_NAME` or `YB_BREW_TYPE` be better than `YB_WHATBREW`?

File: brew-common.sh [hunk 2]
```diff
@@ -31,43 +56,61 @@ fatal() {
...
 }
 
+# Returns the prefix for a new Homebrew/Linuxbrew installation path, based on the current directory,
+# YB_WHATBREW ("homebrew" or "linuxbrew") and YB_BREW_TIMESTAMP (which would be set on demand).
+# The return value is placed in the brew_path_prefix variable in the parent scope.
 get_brew_path_prefix() {
   set_brew_timestamp
-  local brew_dirname="$YB_BREW_DIR_PREFIX-$YB_BREW_TIMESTAMP"
-  local brew_path_prefix="$(realpath .)/$brew_dirname"
+  local brew_dirname=""
```
Comment (ttyusupov) [inline brew-common.sh]: how do we use it?

File: brew-copy.sh [hunk 3]
```diff
@@ -32,21 +32,32 @@ if [[ $# -lt 1 ]]; then
...
+if [[ ! -f $git_sha1_path ]]; then
+  fatal "File '$git_sha1_path' not found"
 fi
+git_sha1=$( cat "$git_sha1_path" )
+get_brew_path_prefix
+BREW_LINK=$brew_path_prefix
+get_fixed_length_path "$BREW_LINK" "$ABS_PATH_LIMIT" "$git_sha1"
+BREW_HOME=$fixed_length_path
+
+log "Existing Homebrew/Linuxbrew installation from $SRC_BREW_HOME to $BREW_HOME"
```
Comment (ttyusupov) [inline brew-copy.sh]: Is this log message missing some words?

File: brew-copy.sh [hunk 4]
```diff
@@ -32,21 +32,32 @@ if [[ $# -lt 1 ]]; then
...
+
+git_sha1_path="$SRC_BREW_HOME/GIT_SHA1"
+if [[ ! -f $git_sha1_path ]]; then
+  fatal "File '$git_sha1_path' not found"
 fi
+git_sha1=$( cat "$git_sha1_path" )
+get_brew_path_prefix
+BREW_LINK=$brew_path_prefix
+get_fixed_length_path "$BREW_LINK" "$ABS_PATH_LIMIT" "$git_sha1"
+BREW_HOME=$fixed_length_path
```
Comment (ttyusupov) [inline brew-copy.sh:52]: Should we check if this directory already exists?

File: build-and-release.sh [hunk 5]
```diff
@@ -54,31 +60,62 @@ export PATH=/usr/local/bin:$PATH
...
 mkdir -p "$brew_dir"
 cd "$brew_dir"
 "$repo_dir/brew-clone-and-build-all.sh"
 
-create_release_cmd=( release create "$tag" -m "Release $tag" )
 has_files=false
-archive_prefix="$brew_dir/$YB_BREW_DIR_PREFIX-$YB_BREW_TIMESTAMP"
+archive_prefix="$brew_dir/$YB_WHATBREW-$YB_BREW_TIMESTAMP"
 log "Looking for .tar.gz files and SHA256 checksum files with prefix: '$archive_prefix'"
+msg_file=/tmp/release_message_${timestamp}_$RANDOM.tmp
```
Comment (ttyusupov) [inline build-and-release.sh]: We can also add process ID `$$` to minimize risk of conflicts.

**Review by ttyusupov — CHANGES_REQUESTED** [review]

File: brew-common.sh (hunk 1 above)
Comment (mbautin) [reply brew-common.sh]: What would these two mean?

File: brew-common.sh (hunk 2 above)
Comment (mbautin) [reply brew-common.sh]: Removed.

**Pull request merged**